### PR TITLE
Don't use System.out in the ProgressListeners

### PR DESCRIPTION
**Pull request by BraisGabin**
~This PR is not ready because I have a problem here. The `DetektProgessListener` is loaded by the ServiceLoader so I can't add the `PrintStream` as a constructor parameter. I need to get it later using `init(Config)` or `init(SetupContext)`. But neigther of them has that reference.~

~Should I add it to any of them? Or should I create a new `init` that provides both `PrintStream`?~

----

This PR is part of a serie of PRs related with how we manage the output.

The main ideas:
- Always use `PrinterStream.println` instead of `println`
- Don't have a default `printerStream`. Because that is the same as use `println`.
- Reduce the noise in out tests. If we always use `PrinterStream` to print output we can use `NullPrinterStream` or other types of `PrinterStream` so we don't print to the stdout when we run tests. With this we can reduce the log size related with tests about 40%.

### CODE REVIEW

**Comment by arturbosch** [comment]
Lets add `out` and `err` to the `SetupContext`. Maybe even move the `info, debug, error` from `ProcessingSettings` functions to it.

**Comment by BraisGabin** [comment]
What do you think about `info`, `debug` and `error` as extension functions of `SetupContext`?

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/DetektProgressListener.kt [hunk 1]
```diff
@@ -1,16 +1,37 @@
 package io.gitlab.arturbosch.detekt.generator
 
 import io.gitlab.arturbosch.detekt.api.FileProcessListener
+import io.gitlab.arturbosch.detekt.api.SetupContext
+import io.gitlab.arturbosch.detekt.api.UnstableApi
 import org.jetbrains.kotlin.psi.KtFile
+import java.io.PrintStream
 
 class DetektProgressListener : FileProcessListener {
```
Comment (arturbosch) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/DetektProgressListener.kt]: Can we remove this copy and just use the real progress listener from cli?

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Runner.kt [hunk 2]
```diff
@@ -13,14 +13,14 @@ class Runner(
     private val outPrinter: PrintStream,
     private val errPrinter: PrintStream
 ) {
-    private val listeners = listOf(DetektProgressListener())
+    private val listeners = listOf(DetektProgressListener(outPrinter))
```
Comment (arturbosch) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Runner.kt]: Lets remove this listener. The generator output is no realy useful.

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/DetektProgressListener.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/DetektProgressListener.kt]: Hm, lets remove this whole class alltogether as the dots are not really useful for the generator :)

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SetupContext.kt [hunk 3]
```diff
@@ -16,4 +17,14 @@ interface SetupContext {
      * Configuration which is used to setup detekt.
      */
     val config: Config
+
+    /**
+     * The channel to log all the output
```
Comment (arturbosch) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SetupContext.kt]:
```suggestion
     * The channel to log all the output.
```

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SetupContext.kt [hunk 4]
```diff
@@ -16,4 +17,14 @@ interface SetupContext {
...
+     */
+    val outPrinter: PrintStream
+
+    /**
+     * The channel to lod all the errors
```
Comment (arturbosch) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/SetupContext.kt]:
```suggestion
     * The channel to log all the errors.
```

**Pull request merged**